### PR TITLE
Don't publish legacy js artifacts

### PR DESCRIPTION
**Pull request by BoD**
Related to #4590

### CODE REVIEW

File: libraries/apollo-gradle-plugin-external/src/main/kotlin/com/apollographql/apollo3/gradle/internal/DefaultApolloExtension.kt [hunk 1]
```diff
@@ -252,6 +254,31 @@ abstract class DefaultApolloExtension(
     }
   }
 
+  @Suppress("ObjectLiteralToLambda")
+  private fun registerCheckLegacyJsTargetTask(): TaskProvider<Task> {
+    return project.tasks.register(ModelNames.checkLegacyJsTarget()) {
```
Comment (martinbonnin) [inline libraries/apollo-gradle-plugin-external/src/main/kotlin/com/apollographql/apollo3/gradle/internal/DefaultApolloExtension.kt:259]: Do we need a task for that? Could we just fail at configuration time, maybe using `afterEvaluate {}` 🙃 ?

File: libraries/apollo-gradle-plugin-external/src/main/kotlin/com/apollographql/apollo3/gradle/internal/DefaultApolloExtension.kt [hunk 2]
```diff
@@ -252,6 +254,31 @@ abstract class DefaultApolloExtension(
...
+        kotlin?.targets?.any { target -> target is KotlinJsTarget && target.irTarget == null } == true
+      })
+      it.outputs.file(outputFile)
+
+      it.doLast(object : Action<Task> {
+        override fun execute(t: Task) {
+          val hasLegacyJsTargetTypes = it.inputs.properties["hasLegacyJsTargetTypes"] as Boolean
+
+          check(!hasLegacyJsTargetTypes) {
+            "Apollo: LEGACY js target is not supported by Apollo, please use IR or BOTH."
```
Comment (martinbonnin) [inline libraries/apollo-gradle-plugin-external/src/main/kotlin/com/apollographql/apollo3/gradle/internal/DefaultApolloExtension.kt:272]: Maybe just omit "BOTH"? Directly point at IR? 

```suggestion
            "Apollo: LEGACY js target is not supported by Apollo, please use IR."
```

Comment (BoD) [reply libraries/apollo-gradle-plugin-external/src/main/kotlin/com/apollographql/apollo3/gradle/internal/DefaultApolloExtension.kt:272]: Good point 👍

File: libraries/apollo-gradle-plugin-external/src/main/kotlin/com/apollographql/apollo3/gradle/internal/DefaultApolloExtension.kt (hunk 1 above)
Comment (BoD) [reply libraries/apollo-gradle-plugin-external/src/main/kotlin/com/apollographql/apollo3/gradle/internal/DefaultApolloExtension.kt:259]: Haha I guess! I'll have a look.

**Review by martinbonnin — APPROVED** [review]

File: libraries/apollo-gradle-plugin-external/src/main/kotlin/com/apollographql/apollo3/gradle/internal/DefaultApolloExtension.kt [hunk 3]
```diff
@@ -184,6 +184,16 @@ abstract class DefaultApolloExtension(
...
+
+      checkForLegacyJsTarget()
+    }
+  }
+
+  private fun checkForLegacyJsTarget() {
+    val kotlin = project.extensions.findByName("kotlin") as? KotlinMultiplatformExtension
+    val hasLegacyJsTarget = kotlin?.targets?.any { target -> target is KotlinJsTarget && target.irTarget == null } == true
+    check(!hasLegacyJsTarget) {
+      "Apollo: LEGACY js target is not supported by Apollo, please use IR."
```
Comment (martinbonnin) [inline libraries/apollo-gradle-plugin-external/src/main/kotlin/com/apollographql/apollo3/gradle/internal/DefaultApolloExtension.kt:196]: We need to add something in the CHANGELOG about this, certainly link to the [Kotlin changelog](https://kotlinlang.org/docs/whatsnew-eap.html#kotlin-js) too

**Pull request merged**